src/domains/invariant.ml

```diff
@@ -40,12 +40,17 @@ let ( || ) = join
 
 
 
-type context = {
+type context1 = {
   path: int option;
   lval: lval option;
 }
 
+type context = {
+  path: int option;
+  lvals: CilLval.Set.t;
+}
+
```

Comment:
Maybe we can have better names here, given that `context` is already over-used as is? Maybe `context_1lval` and `context_lvalset` or something of the sort?

Reply:
Actually, I'll have a look if I can get rid of `context1` altogether: if it's now just used by base analysis internally, then it doesn't need the `path` field at all, but could hopefully just pass the lval around directly.